fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/IndexMetaData.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * IndexMetaData.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2019 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record;
+
+import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordStore;
+import com.apple.foundationdb.tuple.Tuple;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Objects;
+
+/**
+ * A holder class for index-related meta-data. It could include information such as the state of the index ({@link IndexState}),
+ * the latest start time of an index build, and number of attempts. The class also includes additional methods to serialize
+ * ({@link #toBytes(IndexMetaData)}) / de-serialize ({@link #fromBytes(byte[])}) the index meta-data to be stored on / restored from disk.
+ */
+@API(API.Status.MAINTAINED)
+public class IndexMetaData {
+    private static final long newFormatPrefix = Long.MAX_VALUE;
```

Comment:
Constants should always be all caps, like NEW_FORMAT_PREFIX.  However, I don't think you need this at all (see below).